adaptors/mqtt/test/e2e/usability/suite_test.go

```diff
@@ -0,0 +1,235 @@
+package usability_test
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/rancher/octopus/pkg/brain"
+	"github.com/rancher/octopus/pkg/util/object"
+	"github.com/rancher/octopus/test/util/exec"
+
+	"github.com/rancher/octopus/pkg/limb"
+
+	"github.com/rancher/octopus/test/framework"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+
+	dummyv1alpha1 "github.com/rancher/octopus/adaptors/dummy/api/v1alpha1"
+	mqttv1alpha1 "github.com/rancher/octopus/adaptors/mqtt/api/v1alpha1"
+)
+
+var (
+	testCtx       context.Context
+	testCtxCancel context.CancelFunc
+	testCurrDir   string
+	testRootDir   string
+	testEnv       *envtest.Environment
+
+	k8sCfg *rest.Config
+	k8sCli client.Client
+)
+
+func TestOctopus(t *testing.T) {
+	defer GinkgoRecover()
+
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"usability suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func(done Done) {
+	testCtx, testCtxCancel = context.WithCancel(context.Background())
+
+	var err error
+
+	By("bootstrapping test environment")
+	testEnv = &envtest.Environment{
+		UseExistingCluster: pointer.BoolPtr(true),
+	}
+
+	By("creating kubernetes client")
+	var k8sSchema = clientsetscheme.Scheme
+	err = brain.RegisterScheme(k8sSchema)
+	Expect(err).NotTo(HaveOccurred())
+	err = limb.RegisterScheme(k8sSchema)
+	Expect(err).NotTo(HaveOccurred())
+
+	err = registerScheme(k8sSchema)
+	Expect(err).NotTo(HaveOccurred())
+	err = registerDummyScheme(k8sSchema)
+	Expect(err).NotTo(HaveOccurred())
+
+	k8sCfg, err = framework.StartEnv(testRootDir, testEnv, GinkgoWriter)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(k8sCfg).ToNot(BeNil())
+
+	k8sCli, err = client.New(k8sCfg, client.Options{Scheme: k8sSchema})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(k8sCli).ToNot(BeNil())
+
+	installOctopus()
+
+	deployMQTTServer()
+
+	close(done)
+}, 600)
+
+var _ = AfterSuite(func(done Done) {
+	uninstallOctopus()
+
+	By("tearing down test environment")
+	var err = framework.StopEnv(testRootDir, testEnv, GinkgoWriter)
+	Expect(err).ToNot(HaveOccurred())
+
+	if testCtxCancel != nil {
+		testCtxCancel()
+	}
+
+	close(done)
+}, 600)
```

Comment:
I saw that you `deployMQTTServer` in `BeforeSuite`, but why you don't `undeployMQTTServer` in `AfterSuite`?

Reply:
updated